src/main/java/bio/terra/cli/businessobject/User.java

```diff
@@ -172,9 +177,13 @@ public void fetchPetSaCredentials() {
       return;
     }
 
+    // ask SAM for the project-specific pet SA email and persist it on disk
+    SamService samService = new SamService(this, Context.getServer());
+    petSAEmail = samService.getPetSaEmailForProject(googleProjectId);
+    Context.setUser(this);
```

Comment:
Should this method have documentation indicating it sets the current user to active in the global context? I'm not super familiar with the CLI, so that may already be implied.

Reply:
The `Context.setUser` part is just to persist the pet SA email on disk as part of the cached metadata on the current user. It won't ever set a new user as the current one because this method is only called from places where a current user + workspace are already set, but it's a good idea to document that this method expects that. Added comments.